waspc/src/Wasp/Generator/DbGenerator/Auth.hs

```diff
@@ -104,17 +111,41 @@ makeAuthEntity userEntityIdField (userEntityName, _) = case parsePslBody authEnt
           userId    ${userEntityIdTypeText}? @unique
           ${userFieldOnAuthEntityNameText}      ${userEntityNameText}?    @relation(fields: [userId], references: [${userEntityIdFieldName}], onDelete: Cascade)
           ${identitiesFieldOnAuthEntityNameText} ${authIdentityEntityNameText}[]
+          ${sessionsFieldOnAuthEntityNameText}   ${sessionEntityNameText}[]
         |]
 
     authEntityIdTypeText = T.pack authEntityIdType
     userEntityNameText = T.pack userEntityName
     userFieldOnAuthEntityNameText = T.pack userFieldOnAuthEntityName
     authIdentityEntityNameText = T.pack authIdentityEntityName
     identitiesFieldOnAuthEntityNameText = T.pack identitiesFieldOnAuthEntityName
+    sessionsFieldOnAuthEntityNameText = T.pack sessionsFieldOnAuthEntityName
+    sessionEntityNameText = T.pack sessionEntityName
 
     userEntityIdTypeText = T.pack $ show . Psl.Model.Field._type $ userEntityIdField
     userEntityIdFieldName = T.pack $ Psl.Model.Field._name userEntityIdField
 
+makeSessionEntity :: Generator (String, AS.Entity.Entity)
+makeSessionEntity = case parsePslBody sessionEntityPslBody of
+  Left err -> logAndThrowGeneratorError $ GenericGeneratorError $ "Error while generating " ++ sessionEntityName ++ " entity: " ++ show err
+  Right pslBody -> return (sessionEntityName, AS.Entity.makeEntity pslBody)
+  where
+    sessionEntityPslBody =
+      T.unpack
+        [trimming|
+          id        String   @id @unique
+          expiresAt DateTime
+
+          // Needs to be called `userId` for Lucia to be able to create sessions
+          userId String
+          // Needs to be called `auth` for Lucia to be able to get sessions
```

Comment:
Aha, so this `auth` here has nothing to do with our `Auth`, name-wise? If we renamed `Auth` to `Foo`, this would still be `auth`?

Reply:
Lucia uses the model name to find the relation name, if we named it `Foo` it would need to be `foo`.

Reply:
Aha got it! I didn't find that claer from the comment, I would update it.

Also, if I see that correctly, `auth` you hardcoded, but entity name you didn't -> you should also have `auth` dynamically determined then, based on or connected to `authEntityName`